services/lumberjack/src/service.js

```diff
@@ -0,0 +1,255 @@
+        .on('error', logger.error)
+        .start();
+    this._uploadRateTask =
+      createTimeoutTask(this._uploadRate.bind(this), this._uploadInterval)
+        .on('error', logger.error)
+        .start();
+    this._telemetryTask =
+      createTimeoutTask(this._telemetryOverview.bind(this),
+        this._uploadInterval)
+        .on('error', logger.error)
+        .start();
+  }
+
+  /** Get service ping data and write to the database */
+  async _pingTask() {
+    // Get ping data
+    let ping =
+      (await request.get('http://' + this._pingHost + ':' +
+        this._pingPort + '/api/ping')
+        .proto(stats.PingTimes)
+        .timeout(1000)).body;
+
+    // Write data for services
+    for (let endpoint of ping.service_pings) {
+      let { host, port, name } = endpoint;
+      await this._influx.writeMeasurement('ping', [
+        {
+          fields: { apiPing: endpoint.ms },
+          tags: { host, port, name }
+        }], {
+        database: 'lumberjack'
+      });
+    }
+
+    // Write data for devices
+    for (let endpoint of ping.device_pings) {
+      let { host, port, name } = endpoint;
+      await this._influx.writeMeasurement('ping', [
+        {
+          fields: { devicePing: endpoint.ms },
+          tags: { host, port, name }
+        }], {
+        database: 'lumberjack'
+      });
+    }
+  }
+
+  /** Get telemetry upload rate data and write to the database */
+  async _uploadRate() {
+    // Get upload rate
+    let rate =
+      (await request.get('http://' + this._forwardInteropHost + ':' +
+        this._forwardInteropPort + '/api/upload-rate')
+        .proto(stats.InteropUploadRate)
+        .timeout(1000)).body;
+
+    await this._influx.writeMeasurement('upload-rate', [
+      {
+        fields: { t1: rate.total_1, t5: rate.total_5,
+          f1: rate.fresh_1, f5: rate.fresh_5 },
+        tags: { host: this._forwardInteropHost, port: this._forwardInteropPort }
+      }]);
+  }
+
+  /** Get telemetry overview and task queue length and
+  write to the database */
+  async _telemetryOverview() {
+    const OFFLINE = 0;
+    const ONLINE = 1;
+    let gstatus, pstatus;
+
+    //Get telemetry overview
+    let groundTelem =
+      (await request.get('http://' + this._telemetryHost + ':' +
+        this._telemetryPort + '/api/overview')
+        .proto(telemetry.Overview)
+        .timeout(1000)).body;
+
+    //Get length of task queue for the plane
+    let queueLength =
+      (await request.get('http://' + this._telemetryHost + ':' +
+        this._telemetryPort + '/api/queue-length')
+        .timeout(1000)).body;
+
+    //Check if current time is the same or less than the previous
+    //timestate
+    if (groundTelem.time <= gTimes) {
+      gstatus = OFFLINE;
+      pstatus = OFFLINE;
+    } else {
+      gstatus = ONLINE;
+      pstatus = ONLINE;
+    }
+    if (queueLength > this._queueLimit){
+      pstatus = OFFLINE;
```

Comment:
I think calling the connection offline because of a large telemetry queue length is misleading. Offline / Online is really about the antennas sending data and the telemetry service retrieving it and not about if the telemetry service is backlogged. I would say to leave off queue length for now, and in a future PR, graph the telemetry queue length on grafana like everything else. For a user, I would log this as something liked "Pending Telemetry Transactions" instead of queue length, since it's an internal detail.

Reply:
I agree. The operator should make their own discretion on whether a high number of pending transactions equates to disconnected telemetry, since there are other metrics as well (e.g. how frequently telemetry is receiving new data).